client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,198 @@
+		selectedSite && dispatch( fetchSiteDomains( selectedSite.ID ) );
+	}, [ setIsExpandedBasketView, selectedSite, dispatch ] );
+
+	useEffect( () => {
+		//FIXME: This code segment simulates yoast premium already being added when arriving here. To be removed when plugins page is completed.
+		if ( ! isFetchingProducts && products[ 'yoast_premium' ] ) {
+			const yoastProduct = fillInSingleCartItemAttributes(
+				{ product_slug: 'yoast_premium' },
+				products
+			);
+			replaceProductsInCart( [ yoastProduct ] );
+		}
+	}, [ isFetchingProducts, products, replaceProductsInCart ] );
+
+	const redirectToUseDomainFlow = (): void => {
+		const currentUrl = '/plugins/domain';
+		const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ selectedSite?.slug }&source=${ currentUrl }`;
+		page( useYourDomainUrl );
+	};
+
+	const onDomainSelect = async ( suggestion: DomainSuggestions.DomainSuggestion ) => {
+		const { product_slug, domain_name } = suggestion;
+		const domainProduct = {
+			...domainRegistration( {
+				productSlug: product_slug,
+				domain: domain_name,
+				source: 'Marketplace-Yoast-Domain-Upsell',
+			} ),
+			...suggestion,
+		};
+		setDomain( domainProduct );
+
+		//First remove the previously added domain
+		if ( selectedDomainProductUUID ) {
+			await removeProductFromCart( selectedDomainProductUUID );
+			setDomainProductUUID( '' );
+		}
+
+		//Then add the new domain
+		const responseCart = await addProductsToCart( [ domainProduct ] );
+		const productAdded = responseCart.products.find(
+			( { product_slug: added_product_slug } ) => added_product_slug === product_slug
+		);
+		setDomainProductUUID( productAdded?.uuid ?? '' );
+	};
+
+	return (
+		<>
+			<Masterbar>
+				<Item
+					url="#"
+					icon="cross"
+					className="marketplace-domain-upsell__close-button"
+					onClick={ () =>
+						previousPath
+							? page( previousPath )
+							: page( `/plugins/wordpress-seo/${ selectedSite?.slug }` )
+					}
+					tooltip={ translate( 'Close Domain Selection' ) }
+					tipTarget="close"
+				/>
+			</Masterbar>
+			<div
+				className={ classnames( 'marketplace-domain-upsell__root', {
+					'expanded-basket-view': isExpandedBasketView,
+				} ) }
+			>
+				<div className="marketplace-domain-upsell__domain-picker-container">
+					<DomainPicker
+						initialDomainSearch={ siteName }
+						header={ <MarketplaceDomainUpsellHeader /> }
+						analyticsUiAlgo={ ANALYTICS_UI_LOCATION_MARKETPLACE_DOMAIN_SELECTION }
+						analyticsFlowId={ MARKETPLACE_FLOW_ID }
+						onDomainSelect={ onDomainSelect }
+						currentDomain={ selectedDomain as DomainSuggestions.DomainSuggestion }
+						showRecommendationLabel={ false }
+						onUseYourDomainClick={ redirectToUseDomainFlow }
+					/>
+				</div>
+				<div className="marketplace-domain-upsell__shopping-cart-container">
+					<MarketplaceShoppingCart
+						onCheckout={ () => page( '/checkout/' + selectedSite?.slug ) }
```

Comment:
Just like above, we might need to remove the trailing slash if there's no `selectedSite`.

Reply:
Done ! :)